internal/bminventory/inventory.go

```diff
@@ -1702,19 +1676,43 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 	b.eventsHandler.AddEvent(ctx, params.ClusterID, params.NewHostParams.HostID, models.EventSeverityInfo,
 		fmt.Sprintf("Host %s: registered to cluster", hostutil.GetHostnameForMsg(&host)), time.Now())
 
-	nextStepCommand, nextStepArgs := b.formatNextStepRunnerCommand(params.ClusterID.String(), params.NewHostParams.HostID.String())
-
 	hostRegistration := models.HostRegistrationResponse{
-		Host: host,
-		NextStepRunnerCommand: &models.HostRegistrationResponseAO1NextStepRunnerCommand{
-			Command: nextStepCommand,
-			Args:    *nextStepArgs,
-		},
+		Host:                  host,
+		NextStepRunnerCommand: b.generateNextStepRunnerCommand(ctx, &params),
 	}
 
 	return installer.NewRegisterHostCreated().WithPayload(&hostRegistration)
 }
 
+func (b *bareMetalInventory) generateNextStepRunnerCommand(ctx context.Context, params *installer.RegisterHostParams) *models.HostRegistrationResponseAO1NextStepRunnerCommand {
```

Comment:
Why is all this logic here? after all the configuration is static so host api can have this configuration as well no? 
then the only thing that you will need to run is GetNextStepRunnerCommand with host as parameter

Reply:
If I understand correctly, this would require extracting a common configuration type and embedding it in the configuration types of `host` and `bminventory`. Since that's a significant change, I will open a separate ticket for it.